MyXamarinApp.API/Repository/EmployeeRepository.cs

```diff
@@ -23,12 +24,46 @@ public EmployeeRepository(EmployeeContext context, IMapper mapper)
         /// Fetches the list of employees from the Employees table.
         /// </summary>
         /// <returns>List of employees.</returns>
-        public async Task<List<EmployeeModel>> GetAllEmployees()
+        public async Task<List<EmployeeModel>> GetAllEmployees(string searchString, DateTime initialDate, DateTime finalDate, bool status, int currentPage, int pageSize)
         {
-            var employees = await _context.Employees.ToListAsync();
+            List<Employee> employees;
+            
+            if (string.IsNullOrWhiteSpace(searchString))
+            {
+                employees = await _context.Employees.Where(x => x.IsActive.Equals(status))
+                                                    .Skip((currentPage - 1) * pageSize)
+                                                    .Take(pageSize)
+                                                    .ToListAsync();
+            }
+            else
+            {
+                employees = await _context.Employees.Where(x => x.IsActive.Equals(status) &&
+                                                                (x.Name.Contains(searchString) ||
+                                                                x.Role.Contains(searchString) ||
+                                                                x.Salary.ToString().Contains(searchString)))
+                                                                .Skip((currentPage - 1) * pageSize)
+                                                                .Take(pageSize)?
+                                                                .ToListAsync();
```

Comment:
If you do pagination before processing all your filter criteria, you are not returning what the user is expecting.

In our scenario the date filters should be applied first and then followed by pagination